pkg/generators/pull_request.go

```diff
@@ -51,8 +51,8 @@ func (g *PullRequestGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha
 		return nil, EmptyAppSetGeneratorError
 	}
 
-	if appSetGenerator.PullRequest == nil {
-		return nil, EmptyAppSetGeneratorError
+	if appSetGenerator.ApplicationSetTerminalGenerator == nil || appSetGenerator.ApplicationSetTerminalGenerator.PullRequest == nil {
+		return nil, nil
```

Comment:
These checks are inconsistent. Some return `EmptyAppSetGeneratorError` and others return `nil`. I've standardized on `nil` for no particular reason.

These lines should never be hit. As far as I can tell they're just safety checks because the types technically allow this condition.

Reply:
I would revert and return `EmptyAppSetGeneratorError` everywhere when this condition is met.